admin/src/bench/mod.rs

```diff
@@ -30,40 +32,97 @@ static COMMITS: AtomicUsize = AtomicUsize::new(0);
 //static QUERIES: AtomicUsize = AtomicUsize::new(0);
 
 const COMMIT_SIZE: usize = 100;
+// Note on db, this is equal to COMMIT_SIZE.
+// Having smaller allow validating a given amount
+// of keys.
 const COMMIT_PRUNE_SIZE: usize = 90;
 const COMMIT_PRUNE_WINDOW: usize = 2000;
 
-const USAGE: &str = "
-Usage: stress [--readers=<#>][--writers=<#>] [--commits=<l>]
+pub(super) struct BenchAdapter(parity_db::Db);
+
+impl BenchDb for BenchAdapter {
+	type Options = parity_db::Options;
+
+	fn open(path: &std::path::Path) -> Self {
+		BenchAdapter(Db::with_columns(path, 1).unwrap())
+	}
+
+	fn with_options(options: &Self::Options) -> Self {
+		BenchAdapter(Db::open_or_create(options).unwrap())
+	}
+
+	fn get(&self, key: &Key) -> Option<Value> {
+		self.0.get(0, key).unwrap()
+	}
+
+	fn commit<I: IntoIterator<Item=(Key, Option<Value>)>>(&self, tx: I) {
+		self.0.commit(tx.into_iter().map(|(k, v)| (0, k, v))).unwrap()
+	}
+}
+
+/// Stress tests (warning erase db first).
+#[derive(Debug, StructOpt)]
+pub struct Stress {
+	#[structopt(flatten)]
+	pub shared: Shared,
+
+	/// Number of reading threads [default: 4].
+	#[structopt(long)]
+	pub readers: Option<usize>,
+
+	/// Number of writing threads [default: 1].
+	#[structopt(long)]
+	pub writers: Option<usize>,
+
+	/// Start commit index.
+	#[structopt(long)]
+	pub start_commit: Option<usize>,
+
 
-Options:
-	--readers=<#>      Number of reading threads [default: 4].
-	--writers=<#>      Number of writing threads [default: 1].
-	--commits=<n>      Total numbet of inserted commits.
-	--seed=<n>         Random seed used for key generation.
-	--archive          Do not perform deletions.
-	--append           Open an existing database.
-";
+	/// Total number of inserted commits.
+	#[structopt(long)]
+	pub commits: Option<usize>,
+
+	/// Random seed used for key generation.
+	#[structopt(long)]
+	pub seed: Option<u64>,
+
+	/// Open an existing database.
+	#[structopt(long)]
+	pub append: bool,
+
+	/// Do not apply pruning.
+	#[structopt(long)]
+	pub archive: bool,
+
+	/// Do only check (requires append flag).
+	#[structopt(long)]
+	pub check_only: bool,
```

Comment:
Do we actually need that? `--readers=0 --writers=0` has the same effect, doesn't it? I'd rather have `--no-check` that skips final verification.

Reply:
readers 0 and writers 0 seems to stall, I would not really thought of using this combination, so I guess keeping the parameter can make sense. +1 for --no-check.

Reply:
Actually writers 0 could replace check only and readers 0 could replace no-check

Reply:
well check_only without the start parameter is useless, removing it.

Reply:
Rather `--commit=0` should have the same effect